pkg/volume/scaleio/sio_volume.go

```diff
@@ -0,0 +1,407 @@
+	volName    string
+	readOnly   bool
+	options    volume.VolumeOptions
+	configData map[string]string
+
+	volume.MetricsNil
+}
+
+// *******************
+// volume.Volume Impl
+var _ volume.Volume = &sioVolume{}
+
+// GetPath returns the volume directory for the pod
+func (v *sioVolume) GetPath() string {
+	return v.plugin.host.GetPodVolumeDir(
+		v.podUID,
+		kstrings.EscapeQualifiedNameForDisk(sioPluginName),
+		v.volName)
+}
+
+// *************
+// Mounter Impl
+// *************
+var _ volume.Mounter = &sioVolume{}
+
+// CanMount checks to verify that the volume can be mounted prior to Setup.
+// A nil error indicates that the volume is ready for mounitnig.
+func (v *sioVolume) CanMount() error {
+	return nil
+}
+
+func (v *sioVolume) SetUp(fsGroup *int64) error {
+	return v.SetUpAt(v.GetPath(), fsGroup)
+}
+
+// SetUp bind mounts the disk global mount to the volume path.
+func (v *sioVolume) SetUpAt(dir string, fsGroup *int64) error {
+	v.plugin.volumeMtx.LockKey(v.source.VolumeName)
+	defer v.plugin.volumeMtx.UnlockKey(v.source.VolumeName)
+
+	glog.V(4).Info(log("setting up volume %s", v.source.VolumeName))
+	if err := v.setSioMgr(); err != nil {
+		glog.Error(log("setup failed to create scalio manager: %v", err))
+		return err
+	}
+
+	notDevMnt, err := v.plugin.mounter.IsLikelyNotMountPoint(dir)
+	if err != nil && !os.IsNotExist(err) {
+		glog.Error(log("IsLikelyNotMountPoint test failed for dir %v", dir))
+		return err
+	}
+	if !notDevMnt {
+		glog.V(4).Info(log("skipping setup, dir %s already a mount point", v.source.VolumeName))
+		return nil
+	}
+
+	// attach the volume and mount
+	devicePath, err := v.sioMgr.AttachVolume(v.source.VolumeName)
+	if err != nil {
+		glog.Error(log("setup of volume %v failed to attach:  %v", v.source.VolumeName, err))
+		return err
+	}
+	options := []string{}
+	if v.source.ReadOnly {
+		options = append(options, "ro")
+	} else {
+		options = append(options, "rw")
+	}
+
+	glog.V(4).Info(log("mounting device  %s -> %s", devicePath, dir))
+	if err := os.MkdirAll(dir, 0750); err != nil {
+		glog.Error(log("failed to create dir %#v:  %v", dir, err))
+		return err
+	}
+	glog.V(4).Info(log("setup created mount point directory %s", dir))
+
+	diskMounter := &mount.SafeFormatAndMount{
+		Interface: v.plugin.mounter,
+		Runner:    exec.New(),
+	}
+	err = diskMounter.FormatAndMount(
+		devicePath,
+		dir,
+		v.source.FSType,
+		options,
+	)
+
+	if err != nil {
+		glog.Error(log("mount operation failed during setup: %v", err))
+		if err := os.Remove(dir); err != nil && !os.IsNotExist(err) {
+			glog.Error(log("failed to remove dir %s during a failed mount at setup: %v", dir, err))
+			return err
+		}
+		return err
+	}
+
+	glog.V(4).Info(log("successfully setup volume  %s:%s mounted as  %s", v.source.VolumeName, devicePath, dir))
+	return nil
+}
+
+func (v *sioVolume) GetAttributes() volume.Attributes {
+	return volume.Attributes{
+		ReadOnly:        v.source.ReadOnly,
+		Managed:         false,
```

Comment:
@rootfs there's not a clear meaning documented for that field.  Does `managed` mean block device ?